bin/vault_sync_kv_credential.py

```diff
@@ -183,7 +203,14 @@ def stream_events(self, inputs, ew):
             self._logger.debug("{0}: found existing credential".format(input_name))
 
             found_credential = credential_session.storage_passwords[credential_title]
-            if found_credential.content.clear_password != fetched_vault_password:
+
+            found_clear_password = None
+            try:
+                found_clear_password = found_credential.content.clear_password
+            except AttributeError:
+                self._logger.debug("{0}: credential entry has no clear password".format(input_name))
+
+            if found_clear_password != fetched_vault_password:
```

Comment:
This was a previously unhandled possibility of the credential existing, but without a stored password.  Now we check not just for the credential being found, but handling the exception if asking for `clear_password` fails.  There's no way to ask if it exists, only to handle the exception if one occurs.

Reply:
`fetched_vault_password ` is a json dump string, it will never match the `found_clear_password` right?

Reply:
It can match, and will match after the locally stored credential is [updated](https://github.com/splunk/TA-VaultSync/pull/2/files/5d077a8de26f5c0682002f73a09d1cb727b00da6#diff-32e9e19de6edb662cb644a13a81f3f6e9774fa45f4c0aafb2ef7d930306aba68R216).  The "clear password" is just whatever is stored in Splunk's credential store, which sometimes needs to be a JSON string.  This lets us have TA-VaultSync craft that JSON string.

Reply:
ah.. so a clear password is not the actual password. it is more like `found_password_json`

Reply:
Not necessarily.  It's the found cleartext value from the Splunk credential store.  It's not always in JSON.  The ability to store fields from Vault as a JSON string in Splunk was only added because some apps, like those built with the Add-on Builder, store sensitive data as a JSON document in the credential store.  This enables that content to be provided by TA-VaultSync.

I would say it's just as likely that `found_clear_password` is the actual password value, not a JSON document.  Specifically it's just "found cleartext value from the Splunk credential store," and because the field in `passwords.conf` (where it is stored in encrypted form) is called `password`, I named it `found_clear_password`.